### DISCUSSION TITLE
How to give a user permissions on a supervisor namespace?

### DISCUSSION THREAD

**Discussion by rosenhouse** [vmware/govmomi#3623] [Community Help] [answered]
I've looked in the help output and online, and haven't found an answer yet:

  How can I use `govc` to Add Permissions for a user to a supervisor namespace?  i.e. [step 9 in this doc](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-with-tanzu-services-workloads/GUID-177C23C4-ED81-4ADD-89A2-61654C18201B.html).

I can use `govc` to create users, and to create namespaces, but I don't see how to put those two together.

**Comment by dougm** [comment, marked as answer]
There isn't a specific govc command, but you can send any authenticated API request using the [session.login](https://github.com/vmware/govmomi/blob/main/govc/USAGE.md#sessionlogin) command.

Looks like the API for this is: https://developer.broadcom.com/xapis/vsphere-automation-api/latest/vcenter/access/

Here's some examples from what I just tried:

```console
# Example vars
% ns=dev; user=dougm; domain=vsphere.local

# Create access
% govc session.login -r -X POST /api/vcenter/namespaces/instances/$ns/access/$domain/$user?type=USER <<<'{"role": "OWNER"}'

# Get 
% govc session.login -r -X GET /api/vcenter/namespaces/instances/$ns/access/$domain/$user?type=USER
 | jq .
{
  "role": "OWNER",
  "inherited": false
}

% govc session.login -r -X GET /api/vcenter/namespaces/instances/$ns | jq .access_list
[
  {
    "role": "OWNER",
    "subject_type": "USER",
    "subject": "dougm",
    "domain": "vsphere.local"
  }
]

# Update
% govc session.login -r -X PUT /api/vcenter/namespaces/instances/$ns/access/$domain/$user?type=USER <<<'{"role": "VIEW"}'

# Delete
% govc session.login -r -X DELETE /api/vcenter/namespaces/instances/$ns/access/$domain/$user?type=USER
```

**Reply by rosenhouse** [reply]
Thank you, this worked for me!
